src/take.ts

```diff
@@ -1,7 +1,20 @@
-import { _reduceLazy } from "./_reduceLazy";
 import type { LazyEvaluator } from "./pipe";
 import { purry } from "./purry";
 
+type TakeAcc<
+  T extends ReadonlyArray<unknown>,
+  N extends number,
+  Acc extends ReadonlyArray<unknown> = [],
+> = `${N}` extends `-${number}`
+  ? []
+  : Acc["length"] extends N
+    ? Acc
+    : T extends readonly [infer Head, ...infer Tail]
+      ? TakeAcc<Tail, N, [...Acc, Head]>
+      : [...Acc, ...T];
```

Comment:
I'm not sure about this branch. Can we return a narrowed type that would take N into account? See my questions in the tests.